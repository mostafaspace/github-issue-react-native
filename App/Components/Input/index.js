import React from 'react';
import {TextInput, View, Text} from 'react-native';
import PropTypes from 'prop-types';
import Styles from './styles';
import Icon from 'react-native-vector-icons/Ionicons';
import {Colors} from '../../Constants';

export const Input = ({onChangeText, value, label, iconName, error, testID}) => {
    return (
        <View style={Styles.container} testID={testID}>
            <View style={Styles.searchSection}>
                {iconName && <Icon name={iconName} size={28} color={Colors.grey} style={Styles.searchIcon} />}
                <TextInput
                    style={Styles.searchInput}
                    placeholder={label}
                    value={value + ''}
                    onChangeText={onChangeText}
                    underlineColorAndroid={'transparent'}
                    testID={'Input'}
                />
            </View>
            {error !== '' && <Text style={Styles.error}>{error}</Text>}
        </View>
    );
};

Input.propTypes = {
    label: PropTypes.string,
    value: PropTypes.string,
    error: PropTypes.string,
    iconName: PropTypes.string,
    onChangeText: PropTypes.func.isRequired,
};
