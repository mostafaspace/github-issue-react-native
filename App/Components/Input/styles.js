import {StyleSheet, I18nManager} from 'react-native';

export default StyleSheet.create({
    container: {
        width: '100%',
        paddingHorizontal: 16,
    },
    searchSection: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        borderBottomWidth: 1,
        borderBottomColor: '#e7e8ea',
        borderRadius: 16,
        backgroundColor: 'white',
        paddingVertical: 4,
        paddingHorizontal: 16,

        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 1,
        },
        shadowOpacity: 0.18,
        shadowRadius: 1.0,

        elevation: 1,
    },
    searchInput: {
        flex: 1,
        textAlign: I18nManager.isRTL ? 'right' : 'left',
        color: '#222',
        paddingVertical: 16,
    },
    searchIcon: {
        marginVertical: 8,
        marginEnd: 4,
        width: 30,
        height: 30,
    },
    error: {
        fontSize: 10,
        fontWeight: 'bold',
        color: 'red',
        alignSelf: 'flex-start',
        marginHorizontal: 8,
        marginBottom: 4,
        lineHeight: 14,
        paddingTop: 1,
        textAlign: 'left',
    },
});
