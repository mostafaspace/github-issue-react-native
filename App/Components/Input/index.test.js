import React from 'react';
import {render, fireEvent, cleanup} from '@testing-library/react-native';
import {Input} from '../Input';
import renderer from 'react-test-renderer';

afterEach(cleanup);
it('renders component', () => {
    render(<Input onChangeText={() => {}} />);
});

test('test onChangeText', () => {
    const onEventMock = jest.fn();
    const {getByTestId} = render(<Input onChangeText={onEventMock} />);

    fireEvent(getByTestId('Input'), 'onChangeText', 'test text');
    expect(onEventMock).toHaveBeenCalledWith('test text');
});

it('Button matches snapshot', () => {
    const onEventMock = jest.fn();
    const tree = renderer.create(<Input onChangeText={onEventMock} />).toJSON()
    expect(tree).toMatchSnapshot();
});
