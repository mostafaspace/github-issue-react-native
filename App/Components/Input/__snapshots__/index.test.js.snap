// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 16,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomColor": "#e7e8ea",
        "borderBottomWidth": 1,
        "borderRadius": 16,
        "elevation": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 4,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.18,
        "shadowRadius": 1,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[MockFunction]}
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#222",
          "flex": 1,
          "paddingVertical": 16,
          "textAlign": "left",
        }
      }
      testID="Input"
      underlineColorAndroid="transparent"
      value="undefined"
    />
  </View>
  <Text
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "red",
        "fontSize": 10,
        "fontWeight": "bold",
        "lineHeight": 14,
        "marginBottom": 4,
        "marginHorizontal": 8,
        "paddingTop": 1,
        "textAlign": "left",
      }
    }
  />
</View>
`;
